src/Tokenizer/Tokens.php

```diff
@@ -1223,13 +1223,21 @@ public function clearTokenAndMergeSurroundingWhitespace($index)
 
         $prevIndex = $this->getNonEmptySibling($index, -1);
 
-        if ($this[$prevIndex]->isWhitespace()) {
+        if (null === $prevIndex) {
```

Comment:
Can't the changes in `Tokens` class be reverted? Or is this addressing some issue (maybe separate PR then)?

Reply:
the changes are needed for this PR, not sure what extracting these give us?